lib/ai/model/tool.go

```diff
@@ -19,11 +19,17 @@ package model
 import (
 	"context"
 	"fmt"
+	"strconv"
 	"strings"
+	"sync/atomic"
+	"time"
 
+	"github.com/google/uuid"
 	"github.com/gravitational/trace"
 	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
```

Comment:
Can we use v3 here? We want to stop using v2 at some point, as it's not maintained anymore.